test/e2e/specs/undo.test.js

```diff
@@ -10,10 +10,52 @@ import {
 } from '../support/utils';
 
 describe( 'undo', () => {
-	beforeAll( async () => {
+	beforeEach( async () => {
 		await newPost();
 	} );
 
+	it( 'should undo typing after mouse move', async () => {
+		await clickBlockAppender();
+
+		await page.keyboard.type( 'before move' );
+		await page.mouse.down();
+		await page.keyboard.type( ' after move' );
+
+		expect( await getEditedPostContent() ).toMatchSnapshot();
+
+		await pressWithModifier( META_KEY, 'z' );
+
+		expect( await getEditedPostContent() ).toMatchSnapshot();
```

Comment:
We should ideally test caret position here too. Because it resets to the beginning of the field unexpectedly. Apparently this also occurs in the latest plugin release, so not strictly a regression.